apps/explorer/components/Contract/index.tsx

```diff
@@ -40,114 +54,109 @@ export function Contract({ contract, rates, renewedFrom, renewedTo }: Props) {
       {
         label: 'transaction ID',
         entityType: 'transaction',
-        entityValue: contract.transaction_id,
+        entityValue: contract.transactionID,
       },
       {
         label: 'merkle root',
-        value: contract.merkle_root,
+        value: contract.fileMerkleRoot,
       },
       {
         label: 'unlock hash',
-        value: contract.unlock_hash,
+        value: contract.unlockHash,
       },
       {
         label: 'proof confirmed',
         copyable: false,
-        value: String(contract.proof_confirmed),
+        value: String(!!contract.proofTransactionID),
       },
       {
         label: 'negotiation height',
         copyable: false,
-        value: contract.negotiation_height?.toLocaleString() || '-',
+        value:
+          (
+            contract.confirmationIndex?.height ||
```

Comment:
In what scenario does this fallback to the formation height? Might be a good comment to add.

Reply:
I believe the answer is: between when a contract is formed and when it is confirmed but I'm not positive. I know on the frontend side that we're not guaranteed the confirmationIndex. Does that answer track for you?

Reply:
The contracts route should always return a confirmation index. Probably good to keep this check, but we don't index contracts that have not been confirmed.

Reply:
This is what I'm responding to, in terms of Go. The `/contracts` route returns `ExtendedFileContract`, which has this:

```go
ConfirmationIndex         *types.ChainIndex    `json:"confirmationIndex"`
ConfirmationTransactionID *types.TransactionID `json:"confirmationTransactionID"`
```

Reply:
Are you referring to the fact that its a pointer? Although a pointer _could_ be null, the business logic may still be designed to realistically always return a value. @n8maninger is there a way we could make it more clear whether or not the json would actually return a null value?

Reply:
Unclear why it's a pointer for me as well. Only confirmed contracts are indexed, so it should never be nil.

Reply:
I think "could be null" is a good reason to expect in the code that it might be null. Otherwise, it feels to me like we would be hiding an understanding between server and client from consumers. It's quite possible that it's never an issue. But, if the typings match and something goes wrong, it's likely going to be easier to diagnose than if they don't match--from a consumer's perspective. As with anything I write, I'm always happy to defer to greater wisdom here but my view is that it should stay possibly null unless the server-side typing say otherwise.

That said, is `|| formationTxnChainIndex[0].height` the move here or should we display "-" in lieu of the `confirmationIndex`? Looking at this more, we really never reach the "-" case here. We're either getting that formationTxChainIndex or we're error'ing this page, so I could easily delete the OR case here and it would actually be a bit cleaner.

Reply:
Since we know the field will never be null (because only confirmed contracts are indexed), I would lean towards typing it on our end as always defined. If we could also change the type on the Go side to make this fact more clear, that would even better.

@n8maninger is it accurate for us to assume pointers are used because they can be null, and the field would otherwise not be a pointer? Or are they sometimes used in places where they will never be null for other reasons?

Reply:
> Every other instance with pointers I can think of right now is somewhere that can be null and should be explicitly handled.

@n8maninger Does that include confirmationTransactionID?

Reply:
Both `confirmationTransactionID` and `confirmationIndex` are no longer pointers in latest

Reply:
Got this covered now in a downstream PR.